sds/src/secondary_validation/jwt_expiration_checker.rs

```diff
@@ -0,0 +1,130 @@
+use crate::secondary_validation::Validator;
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use chrono::Utc;
+use serde_json::Value as JsonValue;
+
+pub struct JwtExpirationChecker;
+
+const SEGMENTS_COUNT: usize = 3;
+
+impl Validator for JwtExpirationChecker {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        let result = decode_segments(regex_match);
+        if let Ok((_, payload)) = result {
+            let expiration = payload.get("exp");
+            if let Some(exp) = expiration {
+                let exp_int: i64 = exp.as_i64().unwrap();
+                let now = Utc::now().timestamp();
+                exp_int > now
+            } else {
+                // if there is no expiration claim, we consider it as a valid match
+                true
+            }
+        } else {
+            // product decision if an error happend while decoding the JWT shall we consider as a valid match ?
+            // In our case, we consider it as an invalid match (there is no reason to have an invalid JWT outside than having an non valid match)
+            false
+        }
+    }
+}
+
+// This function is an extraction of the decoding part https://github.com/GildedHonour/frank_jwt
+// We don't want to depend on the whole frank_jwt crate for this simple function
+// Our goal is to be able to decode a JWT without checking the signature
+fn decode_segments(encoded_token: &str) -> Result<(JsonValue, JsonValue), ()> {
+    let raw_segments: Vec<&str> = encoded_token.split(".").collect();
+    if raw_segments.len() != SEGMENTS_COUNT {
+        // Invalid JWT header + payload + signature
+        return Err(());
+    }
+
+    let header_segment = raw_segments[0];
+    let payload_segment = raw_segments[1];
+    let result = decode_header_and_payload(header_segment, payload_segment);
+    if let Ok((header, payload)) = result {
+        Ok((header, payload))
+    } else {
+        Err(())
+    }
+}
+
+fn decode_header_and_payload(
+    header_segment: &str,
+    payload_segment: &str,
+) -> Result<(JsonValue, JsonValue), ()> {
+    let b64_to_json = |seg| -> Result<JsonValue, ()> {
+        let res = URL_SAFE_NO_PAD.decode(seg);
+        if let Ok(decoded) = res {
+            serde_json::from_slice(&decoded).map_err(|_| ())
+        } else {
+            Err(())
+        }
+    };
+
+    let header_json = b64_to_json(header_segment)?;
+    let payload_json = b64_to_json(payload_segment)?;
+    Ok((header_json, payload_json))
+}
```

Comment:
I think it would be easier to read if this method way only decoding one segment and the aggregation of segments way done in `decode_segments`. In that sense, `decode_header_and_payload` could probably be replaced by `b64_to_json` (which can be named `decode_segment` in such case). 

As mentioned in the previous comment, this method can return a `Option` instead of a `Result` with empty `Err`. 

Also the method can be composed of chained method to take advantage of the Result/Option framework.
```suggestion
fn decode_segment(segment: &str) -> Option<JsonValue> {
     URL_SAFE_NO_PAD
         .decode(segment)
         .ok()
         .and_then(|decoded| serde_json::from_slice(&decoded).ok())
}
````